File: DAL/Models/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DAL.Models
{
    public class Product
    {
        public int Id { get; set; }
        public int CategoryId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string SerialNumber { get; set; }
        public string PurchaseCost { get; set; }
        public int QuantityInStock { get; set; }
        public DateTime PurchaseDate { get; set; }
        public string Status { get; set; } 
        public DateTime CreatedAt { get; set; } 
        public DateTime UpdatedAt { get; set; } 

        public Category Category { get; set; }
    }
}

File: inventory.client/src/app/shared/models/Product.ts
export interface Product{
    id?:number,
    categoryId:number,
    name:string,
    description:string,
    serialNumber:string,
    purchaseCost:string,
    purchaseDate:Date,
    status:string,
    createdAt?:Date,
    updatedAt:Date,
}